app/models/miq_ae_datastore.rb

```diff
@@ -161,38 +165,50 @@ def self.seed_default_namespace
   def self.reset_to_defaults
     saved_attrs = preserved_attrs_for_domains
 
-    reset_domains_from_legacy_directory
-    reset_domains_from_vmdb_plugins
+    reset_all_domains
 
     restore_attrs_for_domains(saved_attrs)
     reset_default_namespace
     MiqAeDomain.reset_priorities
   end
 
-  def self.reset_domains_from_legacy_directory
+  private_class_method def self.domains
+    legacy_domains + plugin_domains
+  end
+
+  private_class_method def self.default_domains
+    domains.select(&:system?)
+  end
+
+  private_class_method def self.legacy_domains
     domain_files = DATASTORE_DIRECTORY.join('*', MiqAeDomain::DOMAIN_YAML_FILENAME)
-    Dir.glob(domain_files).each do |domain_file|
-      domain_name = File.basename(File.dirname(domain_file))
-      reset_domain(DATASTORE_DIRECTORY, domain_name, Tenant.root_tenant)
+    domains = Dir.glob(domain_files).map do |domain_file|
+      Vmdb::Plugins::AutomateDomain.new(File.dirname(domain_file))
     end
+    filter_domains(domains)
   end
 
-  def self.reset_domains_from_vmdb_plugins
-    datastores_for_reset.each do |domain|
-      reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant)
-    end
+  private_class_method def self.plugin_domains
+    filter_domains(Vmdb::Plugins.automate_domains)
   end
 
-  def self.datastores_for_reset
-    domains = Vmdb::Plugins.automate_domains
+  private_class_method def self.filter_domains(domains)
     return domains unless Rails.env.test? && ENV["AUTOMATE_DOMAINS"]
 
     filter = ENV["AUTOMATE_DOMAINS"].split(",")
     domains.select { |i| filter.include?(i.name) }
   end
 
+  private_class_method def self.domain_names
+    domains.collect(&:name)
+  end
+
   def self.default_domain_names
-    Vmdb::Plugins.system_automate_domains.collect(&:name)
```

Comment:
@d-m-u Aside, I think this is the last caller of Vmdb::Plugins.system_automate_domains, and if so, can you please open a PR to delete it from Vmdb::Plugins in core?

Reply:
https://github.com/ManageIQ/manageiq/pull/18526